plone/namedfile/file.py

```diff
@@ -266,14 +273,24 @@ class NamedImage(NamedFile):
     filename = FieldProperty(INamedFile['filename'])
 
     def __init__(self, data='', contentType='', filename=None):
-        self.contentType, self._width, self._height = getImageInfo(data)
-        self.data = data
         self.filename = filename
+        self._setData(data)
 
         # Allow override of the image sniffer
         if contentType:
             self.contentType = contentType
 
+        if self.contentType in ['image/jpeg', 'image/tiff']:
+            exif_data = get_exif(data)
+            if exif_data is not None:
+                log.debug('Image contains Exif Informations. '
+                          'Test for Image Orientation and Rotate if necessary.'
+                          'Exif Data: %s', exif_data)
+                orientation = exif_data['0th'].get(piexif.ImageIFD.Orientation, 1)
+                if 1 < orientation <= 8:
+                    data = rotate_image(data)
+                self.exif_data = exif_data
```

Comment:
The `__init__` method does nothing more with `data`. Shouldn't this be the same as for NamedBlobImage below?

                if 1 < orientation <= 8:
                    self.data, self._width, self._height, self.exif = rotate_image(self.data)
                else:
                    self.exif = exif_data